cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -1218,26 +1226,44 @@ macro(build_gtest)
         -Wno-ignored-attributes)
   endif()
 
+  if(WIN32)
+    set(GTEST_CMAKE_CXX_FLAGS ${GTEST_CMAKE_CXX_FLAGS} -DGTEST_CREATE_SHARED_LIBRARY=1)
+  endif()
+
   set(GTEST_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/googletest_ep-prefix/src/googletest_ep")
   set(GTEST_INCLUDE_DIR "${GTEST_PREFIX}/include")
+
+  if(MSVC)
```

Comment:
@kou can you confirm if this is needed with MinGW also (so WIN32 instead of MSVC)?

Reply:
I've confirmed. This isn't needed with MinGW.

```text
...
[ RUN      ] TestAllocate.Bitmap
C:/Users/kou/work/arrow/arrow.kou/cpp/src/arrow/testing/gtest_util.cc:120: Failure
Value of: buffer.Equals(expected)
  Actual: false
Expected: true
[  FAILED  ] TestAllocate.Bitmap (0 ms)
...
[  FAILED  ] 1 test, listed below:
[  FAILED  ] TestAllocate.Bitmap

 1 FAILED TEST
...
```